sockets/windows_socket.go

```diff
@@ -0,0 +1,18 @@
+// +build windows
+
+package sockets
+
+import (
+	"net"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// NewWindowsSocket creates a Windows named pipe on the specified path.
+func NewWindowsSocket(addr string, pipeConfig *winio.PipeConfig) (net.Listener, error) {
```

Comment:
I tried to make this symmetrical to unix_socket.go and tcp_socket.go. Should I rename and refactor "NewUnixSocket" and "NewTCPSocket" for this PR?